docs/core/lmdb_dataset_creation.md

```diff
@@ -0,0 +1,251 @@
+```
+
+```{code-cell} ipython3
+system_paths = ["CuCO_adslab.traj"]
+idx = 0
+
+for system in system_paths:
+    # Extract Data object
+    data_objects = read_trajectory_extract_features(a2g, system)
+    initial_struc = data_objects[0]
+    relaxed_struc = data_objects[1]
+    
+    initial_struc.y_init = initial_struc.y # subtract off reference energy, if applicable
+    del initial_struc.y
+    initial_struc.y_relaxed = relaxed_struc.y # subtract off reference energy, if applicable
+    initial_struc.pos_relaxed = relaxed_struc.pos
+    
+    # Filter data if necessary
+    # OCP filters adsorption energies > |10| eV
+    
+    initial_struc.sid = idx  # arbitrary unique identifier 
+    
+    # no neighbor edge case check
+    if initial_struc.edge_index.shape[1] == 0:
+        print("no neighbors", traj_path)
+        continue
+    
+    # Write to LMDB
+    txn = db.begin(write=True)
+    txn.put(f"{idx}".encode("ascii"), pickle.dumps(initial_struc, protocol=-1))
+    txn.commit()
+    db.sync()
+    idx += 1
+
+db.close()
+```
+
+```{code-cell} ipython3
+dataset = SinglePointLmdbDataset({"src": "sample_CuCO.lmdb"})
+len(dataset)
+```
+
+```{code-cell} ipython3
+dataset[0]
+```
+
+## Structure to Energy and Forces (S2EF) LMDBs
+
+S2EF LMDBs utilize the TrajectoryLmdb dataset. This dataset expects a directory of LMDB files. In addition to the attributes defined by AtomsToGraph, the following attributes must be added for the S2EF task:
+
+- tags (optional): 0 - subsurface, 1 - surface, 2 - adsorbate
+- fid: Frame index along the trajcetory
+- sid- sid: Unique system identifier, arbitrary
+
+Additionally, a "length" key must be added to each LMDB file.
+
+As a demo, we will use the above generated data to create an S2EF LMDB dataset
+
+```{code-cell} ipython3
+os.makedirs("s2ef", exist_ok=True)
+db = lmdb.open(
+    "s2ef/sample_CuCO.lmdb",
+    map_size=1099511627776 * 2,
+    subdir=False,
+    meminit=False,
+    map_async=True,
+)
+```
+
+```{code-cell} ipython3
+tags = raw_data[0].get_tags()
+data_objects = a2g.convert_all(raw_data, disable_tqdm=True)
+
+
+for fid, data in tqdm(enumerate(data_objects), total=len(data_objects)):
+    #assign sid
+    data.sid = torch.LongTensor([0])
+    
+    #assign fid
+    data.fid = torch.LongTensor([fid])
+    
+    #assign tags, if available
+    data.tags = torch.LongTensor(tags)
+    
+    # Filter data if necessary
+    # OCP filters adsorption energies > |10| eV and forces > |50| eV/A
+
+    # no neighbor edge case check
+    if data.edge_index.shape[1] == 0:
+        print("no neighbors", traj_path)
+        continue
+
+    txn = db.begin(write=True)
+    txn.put(f"{fid}".encode("ascii"), pickle.dumps(data, protocol=-1))
+    txn.commit()
+    
+txn = db.begin(write=True)
+txn.put(f"length".encode("ascii"), pickle.dumps(len(data_objects), protocol=-1))
+txn.commit()
+
+
+db.sync()
+db.close()
+```
+
+```{code-cell} ipython3
+dataset = TrajectoryLmdbDataset({"src": "s2ef/"})
```

Comment:
Safe to say, for all instances of Single/Trajectory in the docs, let's use LmdbDataset